augur/io/sqlite3.py

```diff
@@ -7,15 +7,31 @@ class Sqlite3Database:
     """Represents a SQLite3 database.
 
     Provides useful methods to manage connections and general I/O.
+
+    If a file is not provided, an in-memory database will be used.
     """
 
-    def __init__(self, file:str):
+    def __init__(self, file:str=''):
```

Comment:
I think we should use `None` instead of the empty string as the default "not present" value. `None` is more conventional in Python for that purpose, esp. for function params.

If you're concerned about making the types more verbose, e.g. `file: Optional[str] = None`, I believe both Mypy and Pyright support options to implicitly treat `file: str = None` in function signatures as if wrapped in `Optional[]`.

This is just one example, but I've noticed a few others.

Reply:
Good catch, done: b3c93f3a810d2558b0bcd1a878531f1b315c6725

I couldn't find any others, but let me know if you see otherwise.